databricks/koalas/frame.py

```diff
@@ -673,6 +673,7 @@ def _reduce_for_stat_function(self, sfun, name, axis=None, numeric_only=True, **
         else:
             # Here we execute with the first 1000 to get the return type.
             # If the records were less than 1000, it uses pandas API directly for a shortcut.
+            numeric_only = None if name in ["sum", "product", "prod"] else numeric_only
```

Comment:
The behavior when `numeric_only=None` is changed for some cases.

Please refer to https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html#consistency-of-dataframe-reductions

Reply:
Can you elabourate a bit more? which functions were changed and how?

Reply:
Btw, let's avoid a logic depending on a specific function in a common function. We should pick a proper value in each function before passing it to the common function.

Reply:
From pandas 1.1.2, the `dtype` of `sum()`, `product()` is `<class 'numpy.object_'>` which cannot inferred in Koalas when `numeric_only=True`.

```python
>>> pdf.sum(axis=1, numeric_only=True).dtype.type
<class 'numpy.object_'>  # object type cannot be inferred in Koalas, raise Exception in `as_spark_type`
>>> pdf.sum(axis=1, numeric_only=None).dtype.type
<class 'numpy.float64'>  # Should use `None` rather than `True` for `sum`, `prod`
```

Reply:
@ueshin 

Sure, let me address it

Reply:
> ```py
> >>> pdf.sum(axis=1, numeric_only=True).dtype.type
> <class 'numpy.object_'>  # object type cannot be inferred in Koalas, raise Exception in `as_spark_type`
> ```

What's the `pdf` like?

I tried in my local:

```py
>>> pd.__version__
'1.2.0'

>>> pdf
   a  b  c
0  1  4  a
1  2  5  b
2  3  6  c
>>> pdf.sum(axis=1, numeric_only=True)
0    5
1    7
2    9
dtype: int64
>>> pdf.sum(axis=1, numeric_only=True).dtype.type
<class 'numpy.int64'>

>>> pdf
     a    b  c
0  1.0  4.0  a
1  2.0  5.0  b
2  3.0  6.0  c
>>> pdf.sum(axis=1, numeric_only=True)
0    5.0
1    7.0
2    9.0
dtype: float64
>>> pdf.sum(axis=1, numeric_only=True).dtype.type
<class 'numpy.float64'>
```

Reply:
Sorry for the late reply.

`pdf` looks like as below.

```python
>> pd.__version__
1.2.0

>>> pdf
   A    B     C      D
0  1  1.0  -6.0   True
1 -2 -2.0  -7.0  False
2  3  3.0  -8.0   True
3 -4 -4.0  -9.0  False
4  5  5.0  10.0  False

>>> pdf.sum(axis=1, numeric_only=True)
0    -3.0
1   -11.0
2    -1.0
3   -17.0
4    20.0
dtype: object

>>> pdf.sum(axis=1, numeric_only=True).dtype.type
<class 'numpy.object_'>
```

Reply:
ah, seems like if there is a bool type column, the return type will be `object`.
I'd suspect it's a bug..